### ISSUE TITLE
Add "docker" command type

### ISSUE THREAD

**Issue by DaoWen** [twosigma/waiter#360]
We should add a `docker` command type in addition to our current `shell` command type.

The version string would be a docker image string (namespace/name:tag). We'd pull the image and run it via Marathon's docker support, or directly with the `docker run` command in the Shell scheduler.

**Comment by dposada** [comment]
Would this play well with the k8s scheduler as well?

**Comment by DaoWen** [comment]
Yup! We could use the provided image in place of the default image we're currently specifying in our ReplicaSet specification.

**Comment by shamsimam** [comment]
Some work to support this command type was done as part of https://github.com/twosigma/waiter/pull/366
